### PR TITLE
Issue #15734: updated JavadocParagraphCheck documentation and added examples for P tag should not precede block-tag

### LINKED ISSUE #15734: JavadocParagraph: update documentation
After merge of https://github.com/checkstyle/checkstyle/issues/15503

We need to update documentation to show extra functionality that this Checks is targeting.

We also need to add such cases in Examples.

### PR DESCRIPTION
**Pull request by Zopsss**
closes #15734 

updated doc and added example for new violation previously added.

2 new examples were added, one with default `allowNewlineParagraph` and other with `allowNewlineParagraph = false`

### CODE REVIEW

**Comment by Zopsss** [comment]
github, generate site

**Comment by Zopsss** [comment]
github, generate site

**Comment by Zopsss** [comment]
GitHub, generate site

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocParagraphCheck.java [hunk 1]
```diff
@@ -40,6 +40,8 @@
  * <li>There is one blank line between each of two paragraphs.</li>
  * <li>Each paragraph but the first has &lt;p&gt; immediately
  * before the first word, with no space after.</li>
+ * <li>First paragraph tag should not precede HTML block-tag,
```
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocParagraphCheck.java:43]: please add link https://www.w3schools.com/html/html_blocks.asp over `HTML block-tag`

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocparagraph/Example3.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    <module name="JavadocParagraph"/>
+  </module>
+</module>
+*/
+package com.puppycrawl.tools.checkstyle.checks.javadoc.javadocparagraph;
+
+// xdoc section -- start
+// violation 6 lines below '<p> tag should not precede HTML block-tag '<h1>''
+// violation 9 lines below '<p> tag should not precede HTML block-tag '<pre>''
+// violation 11 lines below '<p> tag should not precede HTML block-tag '<ul>''
```
Comment (romani) [inline src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocparagraph/Example3.java:13]: 11 is too much, please make javadoc over methods to make javadoc smaller in size.
we can have 2-3 methods to show what we need.
lets max distance from comment to violation as 7.

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocparagraph/Example3.java [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+/**
+ * Some summary.
+ *
+ * <p><h1>P tag is preceding block-tag h1, not allowed</h1>
```
Comment (romani) [inline src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocparagraph/Example3.java:17]: we need to make examples to be same by code and different only by violation comments.
please align all examples by target content and line numbers see rules at https://github.com/checkstyle/checkstyle/issues/13345

**Review by romani — CHANGES_REQUESTED** [review]
items:

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocParagraphCheck.java (hunk 1 above)
Comment (Zopsss) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocParagraphCheck.java]: Done

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocparagraph/Example3.java (hunk 2 above)
Comment (Zopsss) [reply src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocparagraph/Example3.java]: Done. Divided the javadoc into separate parts.

**Comment by Zopsss** [comment]
github, generate site

**Comment by relentless-pursuit** [comment]
github, generate report

**Review by romani — APPROVED** [review]
Ok to merge

**Review by rdiachenko — APPROVED** [review]
lgtm

**Pull request merged**